src/BikramSambat.ts

```diff
@@ -305,4 +305,131 @@ export default class BikramSambat {
   public static getMonthNames(language?: LanguageCode): string[] {
     return NepaliMonthsData.map((month) => month[language ?? 'np'])
   }
+
+  public isSameYear(date: BikramSambat): boolean {
+    if (this.toString() === InvalidDate) {
+      return false
+    }
+    if (this.year === date.getYear()) {
+      return true
+    }
+    return false
+  }
+
+  public isSameMonth(date: BikramSambat): boolean {
+    if (this.toString() === InvalidDate) {
+      return false
+    }
+    if (this.year === date.getYear() && this.month === date.getMonth()) {
+      return true
+    }
+    return false
+  }
+
+  public isSameDay(date: BikramSambat): boolean {
+    if (this.toString() === InvalidDate) {
+      return false
+    }
+    if (
+      this.year === date.getYear() &&
+      this.month === date.getMonth() &&
+      this.day === date.getDay()
+    ) {
+      return true
+    }
+    return false
+  }
+
+  public getWeekStartDate(): BikramSambat {
+    if (this.toString() === InvalidDate) {
+      return this
+    }
+    const currentDate = new BikramSambat(this)
+    const dayOfWeek = currentDate.getDayOfWeek()
+    const startOfWeek = currentDate.addDays(-dayOfWeek)
+    return startOfWeek
+  }
+
+  public getWeekEndDate(): BikramSambat {
+    if (this.toString() === InvalidDate) {
+      return this
+    }
+    const currentDate = new BikramSambat(this)
+    const dayOfWeek = currentDate.getDayOfWeek()
+    const endOfWeek = currentDate.addDays(6 - dayOfWeek)
+    return endOfWeek
+  }
+
+  public isSameWeek(date: BikramSambat): boolean {
+    if (this.toString() === InvalidDate) {
+      return false
+    }
+    const weekStartDate = this.getWeekStartDate()
+    const weekEndDate = this.getWeekEndDate()
+    if (weekStartDate.isAfter(date) && weekEndDate.isBefore(date)) {
+      return true
+    }
+    return false
+  }
+
+  /**
+   * Returns true if the given date is after the current date.
+   * @param date - BikramSambat
+   * @returns boolean
+   * @example
+   * const date = new BikramSambat('2077-01-01')
+   * const date2 = new BikramSambat('2077-01-02')
+   * date.isAfter(date2) // true
+   */
+  public isAfter(date: BikramSambat): boolean {
+    if (
+      this.year === undefined ||
+      this.month === undefined ||
+      this.day === undefined ||
+      date.toString() === InvalidDate
+    ) {
```

Comment:
```suggestion
    if (
      date.toString() === InvalidDate
    ) {
```

toString() method already checks the other condition right?

Reply:
we also need to check for `this` object